spring-rabbit/src/main/java/org/springframework/amqp/rabbit/support/PublisherCallbackChannel.java

```diff
@@ -37,6 +37,8 @@
 
 	String RETURN_CORRELATION_KEY = "spring_listener_return_correlation";
 
+	String RETURNED_MESSAGE_CORRELATION_KEY = "spring_returned_message_correlation";
```

Comment:
Why do we need this extra property?
Maybe the purpose of the existing `RETURN_CORRELATION_KEY` should be extended and its real name change to reflect reality?

Reply:
No; they are different; the first one tells us which template (listener) to call for returned messages; the new one enables us to get a reference to the `PendingConfirm` so we can add the returned message to its correlation data (if present). See the new `pendingReturns` map.